xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-xwiki/src/test/java/org/xwiki/rendering/internal/resolver/RelativeResourceReferenceEntityReferenceResolverTest.java

```diff
@@ -0,0 +1,408 @@
+            (Answer<Boolean>) invocation -> existingDocuments.contains(invocation.getArguments()[0]));
+
+        when(this.defaultEntityProvider.getDefaultReference(EntityType.DOCUMENT))
+            .thenReturn(new EntityReference(DEFAULT_PAGE, EntityType.DOCUMENT));
+    }
+
+    private DocumentResourceReference documentResource(String referenceString, boolean typed)
+    {
+        DocumentResourceReference reference = new DocumentResourceReference(referenceString);
+
+        reference.setTyped(typed);
+
+        return reference;
+    }
+
+    private SpaceResourceReference spaceResource(String referenceString, boolean typed)
+    {
+        SpaceResourceReference reference = new SpaceResourceReference(referenceString);
+
+        reference.setTyped(typed);
+
+        return reference;
+    }
+
+    private AttachmentResourceReference attachmentResource(String referenceString, boolean typed)
+    {
+        AttachmentResourceReference reference = new AttachmentResourceReference(referenceString);
+
+        reference.setTyped(typed);
+
+        return reference;
+    }
+    // Tests
+
+    @Test
+    void resolve()
+    {
+        assertNull(this.resolver.resolve(null, null));
+        assertNull(this.resolver.resolve(new ResourceReference("path", ResourceType.PATH), null));
+    }
+
+    @Test
+    void resolveTypeDocument()
+    {
+        assertEquals(new DocumentReference(WIKI, SPACE, PAGE),
+            this.resolver.resolve(documentResource(WIKI + ':' + SPACE + '.' + PAGE, true), null));
+        assertEquals(PAGE_ENTITY_REFERENCE, this.resolver.resolve(documentResource(SPACE + '.' + PAGE, true), null));
+
+        assertEquals(new EntityReference(PAGE, EntityType.DOCUMENT),
+            this.resolver.resolve(documentResource(PAGE, true), null));
+
+        assertNull(this.resolver.resolve(documentResource("", true), null));
+
+        ResourceReference withBaseReference = documentResource("", true);
+        withBaseReference.addBaseReference(WIKI + ':' + SPACE + '.' + PAGE);
+        assertEquals(new DocumentReference(WIKI, SPACE, PAGE),
+            this.resolver.resolve(withBaseReference, null));
+
+        assertEquals(new DocumentReference(WIKI, SPACE, PAGE), this.resolver
+            .resolve(documentResource("", true), null, new DocumentReference(WIKI, SPACE, PAGE)));
+    }
+
+    @Test
+    void resolveUntypeDocument()
+    {
+        // When the page does not exist
+
+        assertEquals(new DocumentReference(WIKI, List.of(SPACE, PAGE), DEFAULT_PAGE),
+            this.resolver.resolve(documentResource(WIKI + ':' + SPACE + '.' + PAGE, false), null));
+
+        // TODO: check that this assert is ok, we don't get the default page since we don't have any info to properly
+        // resolve the absolute ref (no wiki, no base reference)
+        assertEquals(PAGE_ENTITY_REFERENCE,
+            this.resolver.resolve(documentResource(SPACE + '.' + PAGE, false), null));
```

Comment:
This assertion is probably the first one we should ensure we agree on.

Reply:
This does feel right: when no specific target can be found, an untyped reference is supposed to target a WebHome by default. You don't really need a base reference for that.

Alternatively, you could decide to return an entity reference of type PAGE, in which case you don't need to worry about `WebHome`. Unfortunately, it would probably be a bit dangerous to change this behavior to have the same logic in the absolute resolver.

Reply:
> So it does not feel right?

Yes, sorry, I fixed the comment.